docs/guides/publish.md

```diff
@@ -13,6 +13,19 @@ build it by default. This means that your project may not be ready for distribut
 the effect of declaring a build system in the
 [project concept](../concepts/projects.md#build-systems) documentation.
 
+!!! note
+
+    To avoid accidentally publishing an internal package to PyPI, set this classifier that is valid
```

Comment:
Sorry but I think this still needs an explanatory sentence to start since we're currently in the middle of telling someone how to setup their project _to_ be published. For example:

> If you have internal packages that you do not want to be published, you can mark them as private with a classifier:
>
> ...
>
> uv will still attempt to publish packages with the classifier, but PyPI will reject them.

Reply:
I don't think that level of phrasing matters here

Reply:
Which part?

Is it not a significant security concern in the issue that we attempt to publish the package?

Reply:
Why would a guaranteed-to-fail misconfigured publish attempt be a security concern? If you're really that scared about external sites, then you need to block pypi on a network level as some orgs do, and before that users already shouldn't run `uv publish` locally if you care about such aspects.

Reply:
I mean, the package is still being sent to the host which is contrary to the expectation of a "do not upload" declaration. Their are users in the issue that have brought this up repeatedly. I don't quite understand the pushback on documenting this behavior. If the behavior was more obviously aligned with the expectations, I'd feel differently.

Reply:
Aren't MITM attacks a valid concern? Another concern that comes to mind for me is that you accidentally upload the package to some other index. I just want to be clear about the behavior here since this is basically a "security feature".